File: paimon-python/pypaimon/api/rest_api_names.py
# Licensed to the Apache Software Foundation (ASF) under one
# or more contributor license agreements.  See the NOTICE file
# distributed with this work for additional information
# regarding copyright ownership.  The ASF licenses this file
# to you under the Apache License, Version 2.0 (the
# "License"); you may not use this file except in compliance
# with the License.  You may obtain a copy of the License at
#
#   http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing,
# software distributed under the License is distributed on an
# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
# KIND, either express or implied.  See the License for the
# specific language governing permissions and limitations
# under the License.


class RESTApiNames:
    """Names of the APIs RESTApi calls. Where DLF OpenAPI (DlfNext 2026-01-18) registers an API,
    the name is the one it is registered under. Mirrors the Java RESTApiNames."""

    GET_CONFIG = "GetConfig"

    LIST_DATABASES = "ListDatabases"
    CREATE_DATABASE = "CreateDatabase"
    GET_DATABASE = "GetDatabase"
    ALTER_DATABASE = "AlterDatabase"
    DROP_DATABASE = "DropDatabase"

    LIST_TABLES = "ListTables"
    LIST_TABLE_DETAILS = "ListTableDetails"
    LIST_TABLES_GLOBALLY = "ListTablesGlobally"
    CREATE_TABLE = "CreateTable"
    REGISTER_TABLE = "RegisterTable"
    GET_TABLE = "GetTable"
    GET_TABLE_BY_ID = "GetTableById"
    ALTER_TABLE = "AlterTable"
    REPLACE_TABLE = "ReplaceTable"
    RENAME_TABLE = "RenameTable"
    DROP_TABLE = "DropTable"
    AUTH_TABLE_QUERY = "AuthTableQuery"
    GET_TABLE_TOKEN = "GetTableToken"

    COMMIT_TABLE = "CommitTable"
    GET_TABLE_SNAPSHOT = "GetTableSnapshot"
    GET_VERSION_SNAPSHOT = "GetVersionSnapshot"
    LIST_SNAPSHOTS = "ListSnapshots"
    ROLLBACK_TO_SNAPSHOT = "RollbackToSnapshot"
    GET_SCHEMA = "GetSchema"
    LIST_SCHEMAS = "ListSchemas"
    ROLLBACK_SCHEMA = "RollbackSchema"
    LIST_CONSUMERS = "ListConsumers"
    RESET_CONSUMER = "ResetConsumer"

    LIST_PARTITIONS = "ListPartitions"
    CREATE_PARTITIONS = "CreatePartitions"
    DROP_PARTITIONS = "DropPartitions"
    MARK_DONE_PARTITIONS = "MarkDonePartitions"
    LIST_PARTITIONS_BY_NAMES = "ListPartitionsByNames"
    LIST_PARTITIONS_BY_FILTER = "ListPartitionsByFilter"

    LIST_BRANCHES = "ListBranches"
    CREATE_BRANCH = "CreateBranch"
    DROP_BRANCH = "DropBranch"
    FAST_FORWARD_BRANCH = "FastForwardBranch"
    RENAME_BRANCH = "RenameBranch"

    LIST_TAGS = "ListTags"
    CREATE_TAG = "CreateTag"
    GET_TAG = "GetTag"
    DROP_TAG = "DropTag"

    LIST_VIEWS = "ListViews"
    LIST_VIEW_DETAILS = "ListViewDetails"
    LIST_VIEWS_GLOBALLY = "ListViewsGlobally"
    CREATE_VIEW = "CreateView"
    GET_VIEW = "GetView"
    ALTER_VIEW = "AlterView"
    RENAME_VIEW = "RenameView"
    DROP_VIEW = "DropView"

    LIST_FUNCTIONS = "ListFunctions"
    LIST_FUNCTION_DETAILS = "ListFunctionDetails"
    LIST_FUNCTIONS_GLOBALLY = "ListFunctionsGlobally"
    CREATE_FUNCTION = "CreateFunction"
    GET_FUNCTION = "GetFunction"
    ALTER_FUNCTION = "AlterFunction"
    DROP_FUNCTION = "DropFunction"

    LIST_PERMISSION_ASSIGNMENTS = "ListPermissionAssignments"
    GRANT_PERMISSION_ASSIGNMENT = "GrantPermissionAssignment"
    REVOKE_PERMISSION_ASSIGNMENT = "RevokePermissionAssignment"
    LIST_POLICIES = "ListPolicies"
    CREATE_POLICY = "CreatePolicy"
    DROP_POLICY = "DropPolicy"

    LIST_LABELS = "ListLabels"
    GET_LABEL = "GetLabel"
    UPSERT_LABEL = "UpsertLabel"
    DELETE_LABEL = "DeleteLabel"

    LIST_SEMANTIC_VIEWS = "ListSemanticViews"
    GET_SEMANTIC_VIEW = "GetSemanticView"
    UPSERT_SEMANTIC_VIEW = "UpsertSemanticView"
    DELETE_SEMANTIC_VIEW = "DeleteSemanticView"

File: paimon-api/src/main/java/org/apache/paimon/rest/RESTApiNames.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.apache.paimon.rest;

/**
 * Names of the APIs {@link RESTApi} calls. Where DLF OpenAPI (DlfNext 2026-01-18) registers an API,
 * the name is the one it is registered under.
 */
public final class RESTApiNames {

    public static final String GET_CONFIG = "GetConfig";

    public static final String LIST_DATABASES = "ListDatabases";
    public static final String CREATE_DATABASE = "CreateDatabase";
    public static final String GET_DATABASE = "GetDatabase";
    public static final String ALTER_DATABASE = "AlterDatabase";
    public static final String DROP_DATABASE = "DropDatabase";

    public static final String LIST_TABLES = "ListTables";
    public static final String LIST_TABLE_DETAILS = "ListTableDetails";
    public static final String LIST_TABLES_GLOBALLY = "ListTablesGlobally";
    public static final String CREATE_TABLE = "CreateTable";
    public static final String REGISTER_TABLE = "RegisterTable";
    public static final String GET_TABLE = "GetTable";
    public static final String GET_TABLE_BY_ID = "GetTableById";
    public static final String ALTER_TABLE = "AlterTable";
    public static final String REPLACE_TABLE = "ReplaceTable";
    public static final String RENAME_TABLE = "RenameTable";
    public static final String DROP_TABLE = "DropTable";
    public static final String AUTH_TABLE_QUERY = "AuthTableQuery";
    public static final String GET_TABLE_TOKEN = "GetTableToken";

    public static final String COMMIT_TABLE = "CommitTable";
    public static final String GET_TABLE_SNAPSHOT = "GetTableSnapshot";
    public static final String GET_VERSION_SNAPSHOT = "GetVersionSnapshot";
    public static final String LIST_SNAPSHOTS = "ListSnapshots";
    public static final String ROLLBACK_TO_SNAPSHOT = "RollbackToSnapshot";
    public static final String GET_SCHEMA = "GetSchema";
    public static final String LIST_SCHEMAS = "ListSchemas";
    public static final String ROLLBACK_SCHEMA = "RollbackSchema";
    public static final String LIST_CONSUMERS = "ListConsumers";
    public static final String RESET_CONSUMER = "ResetConsumer";

    public static final String LIST_PARTITIONS = "ListPartitions";
    public static final String CREATE_PARTITIONS = "CreatePartitions";
    public static final String DROP_PARTITIONS = "DropPartitions";
    public static final String MARK_DONE_PARTITIONS = "MarkDonePartitions";
    public static final String LIST_PARTITIONS_BY_NAMES = "ListPartitionsByNames";
    public static final String LIST_PARTITIONS_BY_FILTER = "ListPartitionsByFilter";

    public static final String LIST_BRANCHES = "ListBranches";
    public static final String CREATE_BRANCH = "CreateBranch";
    public static final String DROP_BRANCH = "DropBranch";
    public static final String FAST_FORWARD_BRANCH = "FastForwardBranch";

    public static final String LIST_TAGS = "ListTags";
    public static final String CREATE_TAG = "CreateTag";
    public static final String GET_TAG = "GetTag";
    public static final String DROP_TAG = "DropTag";

    public static final String LIST_VIEWS = "ListViews";
    public static final String LIST_VIEW_DETAILS = "ListViewDetails";
    public static final String LIST_VIEWS_GLOBALLY = "ListViewsGlobally";
    public static final String CREATE_VIEW = "CreateView";
    public static final String GET_VIEW = "GetView";
    public static final String ALTER_VIEW = "AlterView";
    public static final String RENAME_VIEW = "RenameView";
    public static final String DROP_VIEW = "DropView";

    public static final String LIST_FUNCTIONS = "ListFunctions";
    public static final String LIST_FUNCTION_DETAILS = "ListFunctionDetails";
    public static final String LIST_FUNCTIONS_GLOBALLY = "ListFunctionsGlobally";
    public static final String CREATE_FUNCTION = "CreateFunction";
    public static final String GET_FUNCTION = "GetFunction";
    public static final String ALTER_FUNCTION = "AlterFunction";
    public static final String DROP_FUNCTION = "DropFunction";

    public static final String LIST_PERMISSION_ASSIGNMENTS = "ListPermissionAssignments";
    public static final String GRANT_PERMISSION_ASSIGNMENT = "GrantPermissionAssignment";
    public static final String REVOKE_PERMISSION_ASSIGNMENT = "RevokePermissionAssignment";
    public static final String LIST_POLICIES = "ListPolicies";
    public static final String CREATE_POLICY = "CreatePolicy";
    public static final String DROP_POLICY = "DropPolicy";

    public static final String LIST_LABELS = "ListLabels";
    public static final String GET_LABEL = "GetLabel";
    public static final String UPSERT_LABEL = "UpsertLabel";
    public static final String DELETE_LABEL = "DeleteLabel";

    public static final String LIST_SEMANTIC_VIEWS = "ListSemanticViews";
    public static final String GET_SEMANTIC_VIEW = "GetSemanticView";
    public static final String UPSERT_SEMANTIC_VIEW = "UpsertSemanticView";
    public static final String DELETE_SEMANTIC_VIEW = "DeleteSemanticView";

    private RESTApiNames() {}
}
